pvacseq/lib/net_chop.py

```diff
@@ -0,0 +1,122 @@
+        'input',
+        type=argparse.FileType('r'),
+        help="Input filtered file with predicted epitopes"
+    )
+    parser.add_argument(
+        'output',
+        type=argparse.FileType('w'),
+        help="Output FASTA filename for putative neoepitopes"
+    )
+    parser.add_argument(
+        '--method',
+        choices=methods,
+        help="NetChop prediction method to use (\"cterm\" for C term 3.0, \"20s\" for 20S 3.0).  Default: \"cterm\" (C term 3.0)",
+        default='cterm'
+    )
+    parser.add_argument(
+        '--threshold',
+        type=float,
+        help="NetChop prediction threshold.  Default: 0.5",
+        default=0.5
+    )
+    args = parser.parse_args(args_input)
+    chosen_method = str(methods.index(args.method))
+    jobid_searcher = re.compile(r'<!-- jobid: [0-9a-fA-F]*? status: (queued|active)')
+    result_delimiter = re.compile(r'-+')
+    fail_searcher = re.compile(r'(Failed run|Problematic input:)')
+    # staging_file = open('staging.txt', mode='w+')
+    reader = csv.DictReader(args.input, delimiter='\t')
+    writer = csv.DictWriter(
+        args.output,
+         reader.fieldnames+['Best Cleavage Position', 'Best Cleaveage Score'],
+        delimiter='\t',
+        lineterminator='\n'
+    )
+    writer.writeheader()
+    x = 0
+    i=1
+    print("Waiting for results from NetChop... |", end='')
+    sys.stdout.flush()
+    for chunk in split_file(reader, 100):
+        staging_file = tempfile.NamedTemporaryFile(mode='w+')
+        current_buffer = {}
+        for line in chunk:
+            sequence_id = '%010x'%x
+            staging_file.write('>'+sequence_id+'\n')
+            staging_file.write(line['MT Epitope Seq']+'\n')
+            current_buffer[sequence_id] = {k:line[k] for k in line}
+            x+=1
+        staging_file.seek(0)
+        response = requests.post(
+            "http://www.cbs.dtu.dk/cgi-bin/webface2.fcgi",
+            files={'SEQSUB':(staging_file.name, staging_file, 'text/plain')},
+            data = {
+                'configfile':'/usr/opt/www/pub/CBS/services/NetChop-3.1/NetChop.cf',
+                'SEQPASTE':'',
+                'method':chosen_method,
+                'thresh':'%0f'%args.threshold
+            }
+        )
+        while jobid_searcher.search(response.content.decode()):
+            for _ in range(10):
+                sys.stdout.write('\b'+cycle[i%4])
+                sys.stdout.flush()
+                sleep(1)
+                i+=1
+            response = requests.get(response.url)
+        mode=0
+        if fail_searcher.search(response.content.decode()):
+            sys.stdout.write('\b\b')
+            print('Failed!')
+            print("NetChop encountered an error during processing")
+            sys.exit(1)
+        score = -1
+        pos = 0
+        sequence_name = False
+        for line in response.content.decode().split('\n'):
+            if result_delimiter.match(line):
+                mode = (mode+1)%5
```

Comment:
I don't understand what all the manipulation of the mode variable is supposed to achieve. Is there a way to rewrite this in a more intuitive fashion?